klippy/webhooks.py

```diff
@@ -0,0 +1,270 @@
+# Klippy WebHooks registration and server connection
+#
+# Copyright (C) 2020 Eric Callahan <arksine.code@gmail.com>
+#
+# This file may be distributed under the terms of the GNU GPLv3 license
+import logging
+import socket
+import errno
+import json
+import homing
+from klippy import message_startup, message_ready
```

Comment:
FYI, if klippy imports webhooks and webhooks imports klippy, it causes all sorts of havoc.  One solution is to change `printer.get_state_message()` to return both the state category ('startup', 'ready', 'error') along with the state message.